src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -192,3 +254,139 @@ def set_src_position(ast: proto.SrcPosition) -> None:
 def setattr_if_not_none(obj: Any, attr: str, val: Any) -> None:
     if val is not None:
         setattr(obj, attr, val)
+
+
+def _fill_column_ast(ast: proto.Expr, value: ColumnOrLiteral) -> None:
+    """Copy from a Column object's AST, or copy a literal value into an AST expression.
+
+    Args:
+        ast (proto.SpColumnExpr): A previously created Expr() or SpColumnExpr() IR entity intance to be filled
+        value (ColumnOrLiteral): The value from which to populate the provided ast parameter.
+
+    Raises:
+        TypeError: An SpColumnExpr can only be populated from another SpColumnExpr or a valid Literal type
+    """
+    if isinstance(value, snowflake.snowpark.Column):
+        return ast.CopyFrom(value._ast)
+    elif isinstance(value, VALID_PYTHON_TYPES_FOR_LITERAL_VALUE):
+        build_const_from_python_val(value, ast)
+    elif isinstance(value, Expression):
+        pass  # TODO: clean this up
```

Comment:
Would it make sense for this to be some kind of `...` or `raise NotImplementedError()`?